prusti-viper/src/encoder/encoder.rs

```diff
@@ -634,29 +635,31 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         builtin_encoder.encode_builtin_function_name(&function_kind)
     }
 
-    pub fn encode_cast_function_use(&self, src_ty: ty::Ty<'tcx>, dst_ty: ty::Ty<'tcx>) -> String {
+    pub fn encode_cast_function_use(&self, src_ty: ty::Ty<'tcx>, dst_ty: ty::Ty<'tcx>)
+        -> Result<String, PositionlessEncodingError>
```

Comment:
Why not use the `PositionlessEncodingResult` and `EncodingResult` aliases you have defined?